sound/soc/sof/stream-ipc.c

```diff
@@ -52,17 +65,30 @@ int sof_set_stream_data_offset(struct snd_sof_dev *sdev,
 			       size_t posn_offset)
 {
 	struct snd_pcm_substream *substream = sps->substream;
-	struct sof_stream *stream = substream->runtime->private_data;
 
 	/* check if offset is overflow or it is not aligned */
 	if (posn_offset > sdev->stream_box.size ||
 	    posn_offset % sizeof(struct sof_ipc_stream_posn) != 0)
 		return -EINVAL;
 
-	stream->posn_offset = sdev->stream_box.offset + posn_offset;
+	posn_offset += sdev->stream_box.offset;
 
-	dev_dbg(sdev->dev, "pcm: stream dir %d, posn mailbox offset is %zu",
-		substream->stream, stream->posn_offset);
+	if (sps->substream) {
```

Comment:
use substream here and inside the if {} block

Reply:
I completely remove substream and got rid off one variable. similar with msg_data function. I think it makes code neater.